zomes/chat/src/validation.rs

```diff
@@ -55,3 +55,24 @@ pub(crate) fn common_validatation(data: ValidateData) -> ExternResult<ValidateCa
         _ => ValidateCallbackResult::Valid,
     })
 }
+
+pub fn set_read_only_cap_tokens() -> ExternResult<()> {
+    set_global_access("get_active_chatters".to_string())?;
+    set_global_access("list_channels".to_string())?;
+    set_global_access("list_messages".to_string())?;
+    set_global_access("list_all_messages".to_string())?;
+    set_global_access("stats".to_string())?;
+    set_global_access("agent_stats".to_string())?;
+    Ok(())
+}
+
+fn set_global_access(fn_name: String) -> ExternResult<()> {
```

Comment:
This creates separate cap-tokens for each function.  Doesn't it make more sense to have one cap-token that has all the functions in it?  i.e. I think it should take a Vec<String> instead of a string.

Reply:
Ya, sorry that was my mistake. In a very old version of `holochain-rsm` that I used in `holofuel`, when I passed a BTree with multiple functions for one cap token it did not work. It works now.